event/transform.go

```diff
@@ -65,6 +68,7 @@ const (
 
 // TransformMessage is the payload for print and error events
 type TransformMessage struct {
-	Lvl TransformMsgLvl `json:"lvl"`
-	Msg string          `json:"msg"`
+	Lvl  TransformMsgLvl `json:"lvl"`
+	Msg  string          `json:"msg"`
+	Mode string          `json:"mode,omitempty"`
```

Comment:
ah ok, so we're able to do modes on individual messages as well in the end. Not totally sure this'll be necessary, but might be useful for an "ignore all events with mode === `apply`" on the frontend

Reply:
mentioned above, but this is currently only used for `Error` event types, which use the message payload. For the other two message types right now, `Print` & `DatasetPreview`, we don't have a way of plumbing down the mode without adding complexity, so I didn't add them for now.